### DISCUSSION TITLE
Should SSL_read be repeated with same arguments when SSL_ERROR_WANT_READ or SSL_ERROR_WANT_WRITE happens?

### DISCUSSION THREAD

**Discussion by objmagic** [openssl/openssl#25310] [Q&A] [answered]
[SSL_write documentation](https://docs.openssl.org/3.0/man3/SSL_write/) has a warning section:

> When a write function call has to be repeated because [SSL_get_error(3)](https://docs.openssl.org/3.0/man3/SSL_get_error/) returned SSL_ERROR_WANT_READ or SSL_ERROR_WANT_WRITE, it must be repeated with the same arguments.

However, [SSL_read documentation](https://docs.openssl.org/3.0/man3/SSL_read) does not have same warning. Should I assume that it is not necessary to repeat the `SSL_read` call with same arguments? In fact, I found [this diff](https://github.com/openssl/openssl/pull/6240/files#diff-9168ae706505d006747e4094447e78244c3a14557b884c71c5f0e089599dec58) removed the warning paragraph from `SSL_read` documentation. Was this an intentional decision to make it clear to readers it is NOT necessary to repeat `SSL_read` call with same arguments?

**Comment by t8m** [comment, marked as answer]
Yes, SSL_read() does not have to be repeated with the same arguments, unlikely SSL_write() which has this limitation.
